docs/test_development.rst

```diff
@@ -82,6 +82,61 @@ Moreover the unit tests are executed for `each new pull request via Travis
 CI`_.
 
 
+Functional Tests
+====================
+
+Functional API tests are stored in logical chunks in files. Tests are tagged by tags.
+So it is possible to run for example just Ceph tests. All tests should have
+fixture for proper ``setup`` and ``teardown``. All objects created during testing
+should be removed after test run. There should not be any remains after test run.
+
+There are *only* 2 exceptions:
+* CRUD *happy path* tests which are stored in one file where they share
+  object created and deleted during testing tests from file. These tests should run
+  in same order like they are written in the file. CRUD tests **are not** tagged.
+
+* create or import cluster tests which run at the same beginning of testing because
```

Comment:
I would add a note that there is no implemented (and probably won't be) method for cluster removal.